src/app/clusters/closure-control-server/closure-control-server.cpp

```diff
@@ -336,15 +374,65 @@ Status Instance::HandleStop(HandlerContext & ctx, const Commands::Stop::Decodabl
 Status Instance::HandleMoveTo(HandlerContext & ctx, const Commands::MoveTo::DecodableType & commandData)
 {
     MainStateEnum state = GetMainState();
-    VerifyOrReturnValue(CheckCommandStateCompatiblilty(Commands::Stop::Id, state), Status::InvalidInState);
+    VerifyOrReturnValue(CheckCommandStateCompatibility(Commands::Stop::Id, state), Status::InvalidInState);
+
+    // If all command parameters don't have a value, return InvalidCommand
+    VerifyOrReturnValue(commandData.position.HasValue() || commandData.latch.HasValue() || commandData.speed.HasValue(),
+                        Status::InvalidCommand);
+
+    if (commandData.position.HasValue())
+    {
+        VerifyOrReturnError((commandData.position.Value() != TargetPositionEnum::kUnknownEnumValue), Status::ConstraintError);
+        // If Positioning(PS) feature or not, it SHALL return a status code SUCCESS.
+        VerifyOrReturnError(HasFeature(Feature::kPositioning), Status::Success);
+    }
+
+    if (commandData.latch.HasValue())
+    {
+        VerifyOrReturnError(commandData.latch.Value() != TargetLatchEnum::kUnknownEnumValue, Status::ConstraintError);
+
+        // If MotionLatching (LT) feature or not, the server SHALL return a status code SUCCESS
+        VerifyOrReturnError(HasFeature(Feature::kMotionLatching), Status::Success);
+
+        // If manual intervention is required to latch, respond with INVALID_ACTION
+        VerifyOrReturnError(mDelegate.IsLatchManual() == true, Status::InvalidAction);
+    }
+
+    if (commandData.speed.HasValue())
+    {
+        VerifyOrReturnError(commandData.speed.Value() != Globals::ThreeLevelAutoEnum::kUnknownEnumValue, Status::ConstraintError);
+        // If Speed (SP) feature or not, the server SHALL return a status code SUCCESS
+        VerifyOrReturnError(HasFeature(Feature::kSpeed), Status::Success);
+    }
+
+    if (mDelegate.CheckErrorOnDevice())
+    {
+        // If the device is in an error state, set the MainState to Error
+        VerifyOrReturnError(SetMainState(MainStateEnum::kError) == CHIP_NO_ERROR, Status::Failure);
+        // Return Status Failure
+        return Status::Failure;
+    }
+
+    if (mDelegate.IsDeviceReadyToMove())
+    {
+        // If the device is ready to move, set MainState to Moving
```

Comment:
```suggestion
        // If the closure is ready to move, set MainState to Moving
```
In general: please avoid using the word "device", because it has at least 3 different possible meanings in the Matter context and will confuse anyone who reads it.